### PR TITLE
Added legacy validation checklists

### LINKED ISSUE #14: Create NOOP checklists for the validation of the legacy datasets
We need a checklist that just defaults to being `True` for having validation of those datasets that were previously on the legacy DomiNode. 

All these datasets are going to be ingested as-is, since they are assumed to be of enough quality to be made public (they were public before).

This means we need checklists with a single check that just returns `True`.

Create checklists for:

- [ ] vector datasets
- [ ] raster datasets

### PR DESCRIPTION
**Pull request by Samweli**
Adds new checklists for validation of legacy datasets, fixes https://github.com/dominodeorg/dominode-qgis-resources/issues/14.

### CODE REVIEW

File: collections/dominode-resources/processing/legacy.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            'Always returns True'
+        )
+
+    def initAlgorithm(self, config=None):
+        self.addOutput(
+            QgsProcessingOutputString(
+                self.OUTPUT,
+                self.tr('True value')
+            )
+        )
```
Comment (ricardogsilva) [inline collections/dominode-resources/processing/legacy.py]: You are missing the definition of the algorithm's inputs, which is done by calling the algorithm's `self.addParameter()` method.

Check for example the code on `expressiontostringconverter.py` module (which is part of this repo)

File: collections/dominode-resources/processing/legacy.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+from qgis.core import (
+    QgsProcessingAlgorithm,
+    QgsProcessingParameterExpression,
+    QgsProcessingOutputString,
+)
+from qgis.PyQt.QtCore import QCoreApplication
+
+
+class Legacy(QgsProcessingAlgorithm):
```
Comment (ricardogsilva) [inline collections/dominode-resources/processing/legacy.py]: Please choose a more descriptive name that signals that this is a validation step. Something like `NoopValidator` or similar. Also, please change the name of the file to reflect that, to be consistent with how the other algorithm files that exist on this repo are being named.

File: collections/dominode-resources/processing/legacy.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        """
+        Returns the unique algorithm name.
+        """
+        return 'legacy'
+
+    def displayName(self):
+        """
+        Returns the translated algorithm name.
+        """
+        return self.tr('Legacy')
```
Comment (ricardogsilva) [inline collections/dominode-resources/processing/legacy.py]: Please be a bit more verbose on these names to specify clearly what this is (similar to the request for changing the class name above)

File: collections/dominode-resources/checklists/legacy-vector.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  "validation_artifact_type": "dataset",
+  "checks": [
+    {
+      "name": "legacy",
+      "description": "Check for legacy vector datasets.",
+      "guide": "This check will always defaults to True, it has been added for the validation of validation of those datasets that were previously on the legacy DomiNode",
+     "automation": {
+        "algorithm_id": "script:legacy",
+        "artifact_parameter_name": "INPUT",
+        "output_name": "Result"
```
Comment (ricardogsilva) [inline collections/dominode-resources/checklists/legacy-vector.json]: This should be the name you used to define your algorithm's output (check line 14 of the `processing/legacy.py` file below)

File: collections/dominode-resources/checklists/legacy-raster.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  "validation_artifact_type": "dataset",
+  "checks": [
+    {
+      "name": "legacy",
+      "description": "Check for legacy raster datasets.",
+      "guide": "This check will always defaults to True, it hasq been added for the validation of validation of those datasets that were previously on the legacy DomiNode",
+      "automation": {
+        "algorithm_id": "script:legacy",
+        "artifact_parameter_name": "INPUT",
+        "output_name": "Result"
```
Comment (ricardogsilva) [inline collections/dominode-resources/checklists/legacy-raster.json]: This should be the name you used to define your algorithm's output (check line 14 of the `processing/legacy.py` file below)

**Review by ricardogsilva — CHANGES_REQUESTED** [review]

File: collections/dominode-resources/checklists/legacy-raster.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  "description": "A checklist for validating legacy raster datasets",
+  "dataset_type": "raster",
+  "validation_artifact_type": "dataset",
+  "checks": [
+    {
+      "name": "legacy",
+      "description": "Check for legacy raster datasets.",
+      "guide": "This check will always defaults to True, it has been added for the validation of DomiNode legacy raster datasests",
+      "automation": {
+        "algorithm_id": "script:legacy",
```
Comment (ricardogsilva) [inline collections/dominode-resources/checklists/legacy-raster.json]: You need to update this to reflect the new name, which seems to be `script:noopvalidator` (please test it out)

File: collections/dominode-resources/checklists/legacy-vector.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  "description": "A checklist for validating legacy vector datasets",
+  "dataset_type": "vector",
+  "validation_artifact_type": "dataset",
+  "checks": [
+    {
+      "name": "legacy",
+      "description": "Check for legacy vector datasets.",
+      "guide": "This check will always defaults to True, it has been added for the validation of DomiNode legacy vector datasets",
+     "automation": {
+        "algorithm_id": "script:legacy",
```
Comment (ricardogsilva) [inline collections/dominode-resources/checklists/legacy-vector.json]: You need to update this to reflect the new name, which seems to be `script:noopvalidator` (please test it out)

**Review by ricardogsilva — CHANGES_REQUESTED** [review]
Please fix the errors on the automation

**Comment by ricardogsilva** [comment]
@Samweli 

Some additional comment:

- Please remember to remove the `draft` status when you think the work is ready to be merged
- In addition, please explicitly request a review by using the github reviews feature. Otherwise I may miss it :wink: 
- Finally, please provide a brief description for the PR to state what is included, including a reference to the issue that it relates to. If the PR fixes the issue (and in this case it does) please use the appropriate github reference, e.g. fixes #issue-number_

**Comment by Samweli** [comment]
@ricardogsilva I think this PR is ready to be merged.

**Pull request merged**